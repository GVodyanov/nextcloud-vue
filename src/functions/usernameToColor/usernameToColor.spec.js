import usernameToColor from './index'

describe('usernameToColor', () => {
	'use strict'

	const usernames = ['admin', 'foo', 'bar', 'asd', 'wasd', 'a user', 'another user', 'admin@cloud.example.com',
		'Lorelai Taylor',
		'Lillian Wall',
		'Haseeb Stephens',
		'Rae Hope',
		'Santiago Singleton',
		'Daphne Roy',
		'Arham Johnson',
		'Madina Knight',
		'Sid Combs',
		'Zaki Cortes',
		'Ellena Wright Frederic Conway',
		'Alishia Ann Lowry',
		'Idris Mac',
		'Kristi Fisher',
		'Gianluca Hills',
		'Vivienne Jacobs',
		'Brayden Truong',
		'Akeel Robertson']

	usernames.forEach((key) => {
		it(key + ' has the proper color', () => {
			expect(usernameToColor(key)).toMatchSnapshot()
		})
	})

})
