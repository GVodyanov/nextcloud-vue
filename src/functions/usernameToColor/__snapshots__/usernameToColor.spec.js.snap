// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usernameToColor Akeel Robertson has the proper color 1`] = `
Color {
  "b": 164,
  "g": 80,
  "r": 151,
}
`;

exports[`usernameToColor Alishia Ann Lowry has the proper color 1`] = `
Color {
  "b": 109,
  "g": 158,
  "r": 208,
}
`;

exports[`usernameToColor Arham Johnson has the proper color 1`] = `
Color {
  "b": 201,
  "g": 130,
  "r": 0,
}
`;

exports[`usernameToColor Brayden Truong has the proper color 1`] = `
Color {
  "b": 109,
  "g": 158,
  "r": 208,
}
`;

exports[`usernameToColor Daphne Roy has the proper color 1`] = `
Color {
  "b": 164,
  "g": 80,
  "r": 151,
}
`;

exports[`usernameToColor Ellena Wright Frederic Conway has the proper color 1`] = `
Color {
  "b": 133,
  "g": 114,
  "r": 195,
}
`;

exports[`usernameToColor Gianluca Hills has the proper color 1`] = `
Color {
  "b": 97,
  "g": 180,
  "r": 214,
}
`;

exports[`usernameToColor Haseeb Stephens has the proper color 1`] = `
Color {
  "b": 97,
  "g": 180,
  "r": 214,
}
`;

exports[`usernameToColor Idris Mac has the proper color 1`] = `
Color {
  "b": 164,
  "g": 80,
  "r": 151,
}
`;

exports[`usernameToColor Kristi Fisher has the proper color 1`] = `
Color {
  "b": 201,
  "g": 130,
  "r": 0,
}
`;

exports[`usernameToColor Lillian Wall has the proper color 1`] = `
Color {
  "b": 145,
  "g": 92,
  "r": 188,
}
`;

exports[`usernameToColor Lorelai Taylor has the proper color 1`] = `
Color {
  "b": 85,
  "g": 203,
  "r": 221,
}
`;

exports[`usernameToColor Madina Knight has the proper color 1`] = `
Color {
  "b": 164,
  "g": 80,
  "r": 151,
}
`;

exports[`usernameToColor Rae Hope has the proper color 1`] = `
Color {
  "b": 171,
  "g": 90,
  "r": 121,
}
`;

exports[`usernameToColor Santiago Singleton has the proper color 1`] = `
Color {
  "b": 145,
  "g": 92,
  "r": 188,
}
`;

exports[`usernameToColor Sid Combs has the proper color 1`] = `
Color {
  "b": 109,
  "g": 158,
  "r": 208,
}
`;

exports[`usernameToColor Vivienne Jacobs has the proper color 1`] = `
Color {
  "b": 193,
  "g": 120,
  "r": 30,
}
`;

exports[`usernameToColor Zaki Cortes has the proper color 1`] = `
Color {
  "b": 143,
  "g": 166,
  "r": 110,
}
`;

exports[`usernameToColor a user has the proper color 1`] = `
Color {
  "b": 179,
  "g": 100,
  "r": 91,
}
`;

exports[`usernameToColor admin has the proper color 1`] = `
Color {
  "b": 109,
  "g": 158,
  "r": 208,
}
`;

exports[`usernameToColor admin@cloud.example.com has the proper color 1`] = `
Color {
  "b": 164,
  "g": 80,
  "r": 151,
}
`;

exports[`usernameToColor another user has the proper color 1`] = `
Color {
  "b": 85,
  "g": 203,
  "r": 221,
}
`;

exports[`usernameToColor asd has the proper color 1`] = `
Color {
  "b": 181,
  "g": 142,
  "r": 36,
}
`;

exports[`usernameToColor bar has the proper color 1`] = `
Color {
  "b": 201,
  "g": 130,
  "r": 0,
}
`;

exports[`usernameToColor foo has the proper color 1`] = `
Color {
  "b": 109,
  "g": 158,
  "r": 208,
}
`;

exports[`usernameToColor wasd has the proper color 1`] = `
Color {
  "b": 157,
  "g": 70,
  "r": 182,
}
`;
